code/mob/living/life/disability.dm

```diff
@@ -37,12 +37,9 @@
 			if (A.irradiated)
 				//spatial interdictor: mitigate effect of radiation
 				//power expenditure is managed centrally by the interdictor
-				var/interdictor_influence = 0
-				for_by_tcl(IX, /obj/machinery/interdictor)
-					if (IX.radstorm_interdict(owner))
-						interdictor_influence = 1
-						break
-				if(!interdictor_influence)
+				if (owner.hasStatus("spatial_protection"))
```

Comment:
Removing the radstorm_interdict call here means the baseline cost of interdicting a radstorm (independent of number of protected individuals) doesn't apply. My recommended change to preserve the baseline cost would probably be some functionality inside the main process() to apply that baseline cost if there's a currently active mass radiation event and the interdictor applied at least one protection buff on the last tick; to compensate for the per-user cost, it could probably be reduced slightly as well.

Reply:
By the main_process(), do you mean for the radstorm, or interdictor?

Reply:
Ok, so I simply added the radstorm_interdict proc to the end of the interdict proc, and it seems to work. It seems that something is already checking whether or not a radiation storm is going on for the proc. Although as a side effect, it will start draining the increased power when the event is announced, rather than when the radiation hits the station. If this is a problem, I'll try looking for another solution.